server/middleware/users.coffee

```diff
@@ -57,3 +58,54 @@ module.exports =
     yield User.update({ _id: userID }, { $set: { "role": "student" } })
     user = yield User.findById req.user.id
     res.status(200).send(user.toObject({req: req}))
+
+  verifyEmailAddress: wrap (req, res, next) ->
+    user = yield User.findOne({ _id: mongoose.Types.ObjectId(req.params.userID) })
+    [timestamp, hash] = req.params.verificationCode.split(':')
+    if user and req.params.verificationCode is user.verificationCode(timestamp)
+      yield User.update({ _id: user.id }, { emailVerified: true })
+      res.status(200).send({ role: user.toObject().role })
```

Comment:
You can just do `user.get('role')` instead of going through `toObject()`, which makes a copy of the model.